metadata-ingestion/src/datahub/cli/graphql_cli.py

```diff
@@ -0,0 +1,1578 @@
+        if graphql_core_path.exists():
+            return graphql_core_path
+
+    # Fallback: try common relative paths
+    possible_paths = [
+        Path("../../../datahub-graphql-core/src/main/resources"),
+        Path("../../../../datahub-graphql-core/src/main/resources"),
+        Path("datahub-graphql-core/src/main/resources"),
+    ]
+
+    for path in possible_paths:
+        if path.exists():
+            return path.resolve()
+
+    raise FileNotFoundError("Could not find DataHub GraphQL schema files")
+
+
+def _parse_graphql_operations_from_files(
+    custom_schema_path: Optional[str] = None,
+) -> Dict[str, Any]:
+    """Parse operations from DataHub's GraphQL schema files as fallback."""
+    try:
+        if custom_schema_path:
+            schema_path = Path(custom_schema_path)
+            if not schema_path.exists():
+                raise FileNotFoundError(
+                    f"Custom schema path does not exist: {custom_schema_path}"
+                )
+            logger.debug(f"Using custom GraphQL schema path: {schema_path}")
+        else:
+            schema_path = _get_schema_files_path()
+            logger.debug(f"Found GraphQL schema files at: {schema_path}")
+
+        queries = []
+        mutations = []
+
+        # Parse all .graphql files in the directory
+        for graphql_file in schema_path.glob("*.graphql"):
+            content = graphql_file.read_text()
+
+            # Extract queries using regex
+            query_matches = re.finditer(
+                r"extend\s+type\s+Query\s*\{([^}]+)\}|type\s+Query\s*\{([^}]+)\}",
+                content,
+                re.DOTALL | re.IGNORECASE,
+            )
+
+            for match in query_matches:
+                query_content = match.group(1) or match.group(2)
+                operations = _parse_operations_from_content(query_content, "Query")
+                queries.extend(operations)
+
+            # Extract mutations using regex
+            mutation_matches = re.finditer(
+                r"extend\s+type\s+Mutation\s*\{([^}]+)\}|type\s+Mutation\s*\{([^}]+)\}",
+                content,
+                re.DOTALL | re.IGNORECASE,
+            )
+
+            for match in mutation_matches:
+                mutation_content = match.group(1) or match.group(2)
+                operations = _parse_operations_from_content(
+                    mutation_content, "Mutation"
+                )
+                mutations.extend(operations)
+
+        logger.debug(
+            f"Parsed {len(queries)} queries and {len(mutations)} mutations from schema files"
+        )
+
+        return {
+            "queryType": {"fields": queries} if queries else None,
+            "mutationType": {"fields": mutations} if mutations else None,
+        }
+
+    except Exception as e:
+        logger.debug(f"Failed to parse GraphQL schema files: {e}")
+        # Return a minimal fallback with common operations
+        return _get_minimal_fallback_operations()
```

Comment:
IMO if we fail the parsing, we should just fail and no return the minimal fallback operations.